Data-Structures/HashTable/HashTable.js

```diff
@@ -0,0 +1,105 @@
+export default class HashTable {
+  constructor(limit = 100) {
+    // Initialize the storage and limit variables
+    this.storage = new Array(limit)
+    this.limit = limit
+  }
+
+  // Hash function:
+  _hash(key, max) {
+    // Initialize the hash variable to 0
+    let hash = 0
+    // Iterate through the key and add the character code at each iteration to the hash variable
+    for (let i = 0; i < key.length; i++) hash += key.charCodeAt(i)
+    // Return the hash modulo the max
+    return hash % max
+  }
+
+  // Insert a key-value pair into the hash table
+  set(key, value) {
+    // Hash the key
+    const index = this._hash(key, this.limit)
+    // If the index is empty, insert the key-value pair and create a bucket
+    if (this.storage[index] === undefined) this.storage[index] = [[key, value]]
+    else {
+      //  If the index is not empty, iterate through the bucket (collision handling)
+      let inserted = false
+      for (let i = 0; i < this.storage[index].length; i++) {
+        if (this.storage[index][i][0] === key) {
+          // If the key exists, update the value
+          this.storage[index][i][1] = value
+          inserted = true
+        }
+      }
+      // If the key does not exist, insert the key-value pair
+      if (inserted === false) this.storage[index].push([key, value])
+    }
+  }
+
+  // Get a value from the hash table
+  get(key) {
+    // Hash the key
+    const index = this._hash(key, this.limit)
+    // If the index is empty, return undefined
+    if (this.storage[index] === undefined) return undefined
+    else {
+      // If the index is not empty, iterate through the bucket
+      for (let i = 0; i < this.storage[index].length; i++) {
+        // If the key exists, return the value
+        if (this.storage[index][i][0] === key) return this.storage[index][i][1]
+      }
+    }
+  }
+
+  // Remove a key-value pair from the hash table
+  remove(key) {
+    // Hash the key
+    const index = this._hash(key, this.limit)
+    // Check if the bucket exists
+    if (this.storage[index]) {
+      // If the key matches the key at the index and there is only one item in the bucket, delete the bucket
+      if (
+        this.storage[index].length === 1 &&
+        this.storage[index][0][0] === key
+      ) {
+        delete this.storage[index]
+      } else {
+        // If the index is not empty, iterate through the bucket
+        for (let i = 0; i < this.storage[index].length; i++) {
+          // If the key exists, delete the key-value pair
+          if (this.storage[index][i][0] === key) delete this.storage[index][i]
```

Comment:
You're creating an array with holes here. I think this is rather dirty; it results in suboptimal time complexities (deleted items negatively affecting the time it takes to search a bucket).

It would be cleaner to just take an entry from the end, swap the entry to be deleted with that, then pop the entry from the end.